guide/qip/qip-processor.rst

```diff
@@ -101,47 +181,222 @@ To let it find the optimal pulses, we need to give the parameters for :func:`qut
+In order to simulate quantum circuits at the level of time evolution.
+We need to first compile the circuit into the Hamiltonian model, i.e.
+the control pulses.
+Hence each :class:`~qutip.qip.device.Processor` has a corresponding 
+:class:`~qutip.qip.compiler.GateCompiler` class.
+The compiler takes a :class:`~qutip.qip.circuit.QubitCircuit`
+and returns the compiled ``tlist`` and ``coeffs``.
+It is called implicitly when calling the method
+:class:`~qutip.qip.device.Processor.run_state`.
 
-The first example is a processor with one qubit under rotation around the z-axis and relaxation time :math:`T_2=5`. We measure the population of the :math:`\left| + \right\rangle` state and observe the Ramsey signal:
+.. testcode::
 
-.. image:: /gallery/qutip_examples/qip/images/sphx_glr_plot_qip_relaxation_001.png
+    from qutip.qip.compiler import SpinChainCompiler
+    qc = QubitCircuit(2)
+    qc.add_gate("X", targets=0)
+    qc.add_gate("X", targets=1)
+
+    processor = LinearSpinChain(2)
+    compiler = SpinChainCompiler(
+        2, params=processor.params, pulse_dict=processor.pulse_dict)
+    resolved_qc = qc.resolve_gates(["RX", "RZ", "ISWAP"])
+    tlists, coeffs = compiler.compile(resolved_qc)
+    print(tlists)
+    print(coeffs)
+
+**Output**
+
+.. testoutput::
+    :options: +NORMALIZE_WHITESPACE
+
+    [array([0., 1.]), array([0., 1., 2.]), None, None, None]
+    [array([1.57079633]), array([0.        , 1.57079633]), None, None, None]
+
+Here we first use :func:`~qutip.qip.circuit.QubitCircuit.resolve_gates`
+to decompose the X gate to its natural gate on Spin Chain model,
+the rotation over X-axis.
+We pass the hardware parameters of the :class:`~qutip.qip.device.SpinChain `` model, ``processor.params``, as well as a map between the pulse name and pulse index ``pulse_dict`` to the compiler.
+The later one allows one to address the pulse more conveniently in the compiler.
+
+The compiler returns a list of `tlist` and `coeff`, corresponding to each pulse.
+The first pulse starts from `t=0` and ends at `t=1`, with the strengh :math:`\pi/2`.
+The second one is turned on from `t=1` to `t=2` with the same strength.
+The compiled pulse here is different from what is shown in the plot
+in the previous subsection because the scheduler is turned off by default.
```

Comment:
I think most things need double backticks here (except the maths)

Reply:
Yes I always forget that.... Get confused with markdown.

Reply:
yeah, me too